controller/hd44780/hd44780.go

```diff
@@ -0,0 +1,641 @@
+/*
+Package hd44780 allows controlling an HD44780-compatible character LCD
+controller. Currently the library is write-only and does not support
+reading from the display controller.
+
+Resources
+
+This library is based three other HD44780 libraries:
+	Adafruit	https://github.com/adafruit/Adafruit-Raspberry-Pi-Python-Code/blob/master/Adafruit_CharLCD/Adafruit_CharLCD.py
+	hwio		https://github.com/mrmorphic/hwio/blob/master/devices/hd44780/hd44780_i2c.go
+	LiquidCrystal	https://github.com/arduino/Arduino/blob/master/libraries/LiquidCrystal/LiquidCrystal.cpp
+*/
+package hd44780
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kidoman/embd"
+)
+
+type entryMode byte
+type displayMode byte
+type functionMode byte
+
+// RowAddress defines the DDRAM address of the first column of each row, up to 4 rows.
+type RowAddress [4]byte
+
+var (
+	RowAddress16Col RowAddress = [4]byte{0x00, 0x40, 0x10, 0x50} // row addresses for 16-column displays
+	RowAddress20Col RowAddress = [4]byte{0x00, 0x40, 0x14, 0x54} // row addresses for 20-column displays
+)
+
+// BacklightPolarity defines the polarity of the backlight switch, either positive or negative
+type BacklightPolarity bool
+
+const (
+	Negative BacklightPolarity = false
```

Comment:
Are these deliberately exported? (Same question for the other exported symbols.) If yes, then could we had some documentation.